test/integration/helpers/createCompiler.ts

```diff
@@ -0,0 +1,162 @@
+import webpack from 'webpack';
+import VueLoaderPlugin from 'vue-loader/lib/plugin';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as copyDir from 'copy-dir';
+import * as rimraf from 'rimraf';
+import { ForkTsCheckerWebpackPlugin, webpackMajorVersion } from './';
+
+if (!beforeAll || !afterAll) {
+  throw new Error('file should not be included outside of jest!');
+}
+
+const baseTmpDir = path.resolve(__dirname, '../../tmp');
+let tmpDirParent: string;
+let lastInstantiatedPlugin: ForkTsCheckerWebpackPlugin | undefined;
+
+beforeAll(function init() {
+  if (!fs.existsSync(baseTmpDir)) fs.mkdirSync(baseTmpDir);
+
+  tmpDirParent = fs.mkdtempSync(
+    path.join(baseTmpDir, 'fork-ts-checker-webpack-plugin-test')
+  );
+});
+
+afterAll(function cleanUp() {
+  rimraf.sync(tmpDirParent);
+});
+
+afterEach(function killPlugin() {
+  if (lastInstantiatedPlugin && lastInstantiatedPlugin['service']) {
+    lastInstantiatedPlugin['killService']();
+  }
+});
+
+export interface CreateCompilerOptions {
+  entryPoint: string;
+  context: string;
+  happyPackMode: boolean;
+  pluginOptions: Partial<ForkTsCheckerWebpackPlugin.Options>;
+  prepareWebpackConfig(config: webpack.Configuration): webpack.Configuration;
+  nodeRequires: string[];
+  normalizePaths: boolean;
+}
+
+const defaultOptions: Partial<ForkTsCheckerWebpackPlugin.Options> = {
+  silent: true
+};
+
+type CreateCompilerResults = {
+  compiler: webpack.Compiler;
+  readonly compilerConfig: webpack.Configuration;
+  plugin: ForkTsCheckerWebpackPlugin;
+  contextDir: string;
+  outDir: string;
+  tmpDir: string;
+};
+
+function prepareDirectory({ context }: { context: string }) {
```

Comment:
Could you document functions like `prepareDirectory`? The name `prepareDirectory` doesn't tell why it prepares the directory, what is the context :)

Reply:
yup, good idea :)